src/js/player.js

```diff
@@ -1255,6 +1266,24 @@ class Player extends Component {
     return;
   }
 
+  middlewareIterator(method) {
+    return (value, mw) => {
+      if (mw[method]) {
+        return mw[method](value);
+      }
+
+      return value;
+    };
+  }
+
+  middlewareSet(method, arg) {
+    return this.middleware_.reduce(this.middlewareIterator(method), arg);
+  }
+
+  middlewareGet(method) {
+    return this.middleware_.reduceRight(this.middlewareIterator(method), undefined);
+  }
```

Comment:
Right, I was assuming that `this.middleware_` would be passed in.

The `player.js` file is big enough as it is. Seems like iterating over a list of middlewares should be in the middlewares file instead.